### PR TITLE
Fix multiline #: annotations losing parameters

### LINKED ISSUE #3: Multiline #: annotations lose parameters
## Problem

Multiline `#:` annotations lose parameters during parsing. Only the closing line is captured.

### Input

```ruby
class MultilineTest
  #: (
  #:   name: String,
  #:   age: Integer,
  #:   ?email: String?
  #: ) -> Hash[Symbol, untyped]
  def self.call(name:, age:, email: nil)
    { name: name, age: age }
  end
end
```

### Expected output

```rbs
class MultilineTest
  def self.call: (name: String, age: Integer, ?email: String?) -> Hash[Symbol, untyped]
end
```

### Actual output (v0.3.1)

```rbs
class MultilineTest
  def self.call: ) -> Hash[Symbol, untyped]
end
```

The opening `(` and all parameter lines are dropped. Only the closing `) -> return_type` is emitted.

## Why this matters

Long method signatures with many keyword parameters become unreadable on a single line. Multiline annotations are the standard formatting convention in both `rbs_inline` and `@rbs` style annotations.

### PR DESCRIPTION
**Pull request by AndyGauge**
## Summary

- Multiline `#:` annotations now accumulate continuation lines using the existing `is_balanced()` delimiter check, instead of overwriting with only the last line
- Consecutive balanced (complete) single-line `#:` annotations retain last-wins behavior
- Adds 3 new tests covering multiline instance methods, class methods, and the non-merge edge case

Closes #3

## Test plan
- [x] `cargo test` — all 27 tests pass (3 new multiline annotation tests)
- [x] `cargo build --release` — clean release build
- [ ] Verify with a real project that has multiline `#:` annotations

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/transpiler.rs [hunk 1]
```diff
@@ -640,6 +652,88 @@ end
         assert!(result.contains("def validate: () -> void"), "Missing validate, got: {}", result);
     }
 
+    #[test]
+    fn test_multiline_annotation() {
+        let test_file = Path::new("/tmp/test_multiline_annotation.rb");
+        fs::write(
```
[the remark this answers is not in the document]
Comment (AndyGauge) [reply src/transpiler.rs:658]: Valid concern, but all 24 existing tests in this file use hardcoded `/tmp/` paths. Migrating to `tempfile` should be a separate cleanup PR to keep this fix focused.

File: src/transpiler.rs [hunk 2]
```diff
@@ -640,6 +652,88 @@ end
...
+  end
+end
+",
+        )
+        .unwrap();
+
+        let mut transpiler = SentinelTranspiler::new();
+        let result = transpiler.transpile_file(test_file).unwrap();
+        assert!(
+            result.contains("def self.call: ( name: String, age: Integer, ?email: String? ) -> Hash[Symbol, untyped]"),
```
[the remark this answers is not in the document]
Comment (AndyGauge) [reply src/transpiler.rs:678]: Intentionally kept precise — every other test in this file asserts on the exact output string. If the join whitespace changes we want the test to catch it. Substring checks would pass on malformed output missing parentheses entirely.

File: src/transpiler.rs [hunk 3]
```diff
@@ -640,6 +652,88 @@ end
...
+            result
+        );
+    }
+
+    #[test]
+    fn test_multiline_annotation_instance_method() {
+        let test_file = Path::new("/tmp/test_multiline_instance.rb");
+        fs::write(
+            test_file,
+            "\
```
[the remark this answers is not in the document]
Comment (AndyGauge) [reply src/transpiler.rs:689]: Good catch — added `test_multiline_annotation_class_self_block` covering multiline `#:` inside `class << self`.
